specific/cmo_sale/models/sale.py

```diff
@@ -93,11 +141,41 @@ def _compute_project_number(self):
 
     @api.model
     def _default_covenant(self):
-        convenants = self.env['sale.convenant.description'].search([
-            ['active', '=', True],
-        ])
-        if convenants:
-            return convenants[0].description
+        Description = self.env['sale.covenant.description']
+        covenants = Description.search([('active', '=', True), ])
+        return covenants and covenants[0].description or False
+
+    @api.multi
+    @api.constrains('order_line')
+    def _constrains_order_line(self):
+        self.ensure_one()
+        if not self.order_line:
+            raise ValidationError("Must have at least 1 order line!")
+        else:
+            for line in self.order_line:
+                if ((line.price_unit <= 0) or (line.product_uom_qty <= 0)) and \
+                   (line.order_lines_group == 'before'):
+                    raise ValidationError(
+                        "Unit Price and Quantity in order \
+                        line must more than zero !"
+                    )
+
+    @api.multi
+    def _get_amount_by_custom_group(self, custom_group):
+        self.ensure_one()
```

Comment:
Note: This one is OK to use self.ensure_one(), as we know exactly that the caller is only 1 object.
(I try to compare with the case that we can use multi)